code/modules/spacepod/spacepod_parts.dm

```diff
@@ -0,0 +1,174 @@
+	you can find where this part should be by looking at the current direction of the current part and applying the link_angle
+	the link_angle is the angle between the part's direction and its following part, which is the current part's link_to
+	the code works by going in a loop - each part is capable of starting a loop by checking for the part after it, and that part checking, and so on
+	this 4-part loop, starting from any part of the frame, can determine if all the parts are properly in place and aligned
+	it also checks that each part is unique, and that all the parts are there for the spacepod itself
+	*/
+	var/neededparts = list(/obj/item/pod_parts/pod_frame/aft_port, /obj/item/pod_parts/pod_frame/aft_starboard, /obj/item/pod_parts/pod_frame/fore_port, /obj/item/pod_parts/pod_frame/fore_starboard)
+	var/turf/T
+	var/obj/item/pod_parts/pod_frame/linked
+	var/obj/item/pod_parts/pod_frame/pointer
+	var/list/connectedparts =  list()
+	neededparts -= src
+	linked = src
+	for(var/i in 1 to 4)
+		T = get_turf(get_step(linked, turn(linked.dir, -linked.link_angle))) //get the next place that we want to look at
+		var/link_to_in_t = locate(linked.link_to) in T
+		if(link_to_in_t)
+			pointer = link_to_in_t
+		if(istype(pointer, linked.link_to) && pointer.dir == linked.dir && pointer.anchored)
+			if(!(pointer in connectedparts))
+				connectedparts += pointer
+			linked = pointer
+			pointer = null
+	if(connectedparts.len < 4)
+		return
+	for(var/i in 1 to 4)
+		var/obj/item/pod_parts/pod_frame/F = connectedparts[i]
+		if(F.type in neededparts) //if one of the items can be founded in neededparts
+			neededparts -= F.type
+		else //because neededparts has 4 distinct items, this must be called if theyre not all in place and wrenched
+			return
+	return connectedparts
+
+/obj/item/pod_parts/pod_frame/attackby(var/obj/item/O, mob/user)
+	. = ..()
+	if(istype(O, /obj/item/stack/rods))
+		var/obj/item/stack/rods/R = O
+		var/list/linkedparts = find_square()
+		if(!linkedparts)
+			to_chat(user, "<span class='rose'>You cannot assemble a pod frame because you do not have the necessary assembly. Did you ensure to align the frame properly and wrench all of it down?</span>")
+			return
+		var/obj/structure/spacepod_frame/pod = new /obj/structure/spacepod_frame(loc)
+		pod.dir = dir
+		to_chat(user, "<span class='notice'>You strut the pod frame together.</span>")
+		R.use(10)
+		for(var/obj/item/pod_parts/pod_frame/F in linkedparts)
+			if(NORTH == turn(F.dir, -F.link_angle)) //if the part links north during construction, as the bottom left part always does
+				pod.loc = F.loc
+			qdel(F)
+		playsound(get_turf(src), O.usesound, 50, 1)
+	if(istype(O, /obj/item/wrench))
+		to_chat(user, "<span class='notice'>You [!anchored ? "secure \the [src] in place."  : "remove the securing bolts."]</span>")
+		anchored = !anchored
+		density = anchored
+		playsound(get_turf(src), O.usesound, 50, 1)
+
+/obj/item/pod_parts/pod_frame/verb/rotate()
```

Comment:
requesting this also be possible to be done via altclick for consistency.

Reply:
done